src/sentry/runner/initializer.py

```diff
@@ -201,13 +201,19 @@ def apply_legacy_settings(settings):
                       "Use SENTRY_OPTIONS instead, key 'system.admin-email'", DeprecationWarning)
         settings.SENTRY_OPTIONS['system.admin-email'] = settings.SENTRY_ADMIN_EMAIL
 
-    if settings.SENTRY_URL_PREFIX in ('', 'http://sentry.example.com') and not settings.DEBUG:
-        # Maybe also point to a piece of documentation for more information?
-        # This directly coincides with users getting the awkward
-        # `ALLOWED_HOSTS` exception.
-        show_big_error('SENTRY_URL_PREFIX is not configured')
-        # Set `ALLOWED_HOSTS` to the catch-all so it works
-        settings.ALLOWED_HOSTS = ['*']
+    if not settings.SENTRY_OPTIONS.get('system.url-prefix') and hasattr(settings, 'SENTRY_URL_PREFIX'):
+        import warnings
+        warnings.warn('SENTRY_URL_PREFIX is deprecated.'
+                      "Use SENTRY_OPTIONS instead, key 'system.url-prefix'", DeprecationWarning)
+        settings.SENTRY_OPTIONS['system.url-prefix'] = settings.SENTRY_URL_PREFIX
+
+    if not hasattr(settings, 'SENTRY_URL_PREFIX'):
+        from sentry import options
+        url_prefix = options.get('system.url-prefix', silent=True)
+        if not url_prefix:
+            # HACK: We need to have some value here for backwards compatibility
+            url_prefix = 'http://sentry.example.com'
+        settings.SENTRY_URL_PREFIX = url_prefix
```

Comment:
It won't require them to fill it out in the form though, since we already have the value set. So that field will be missing entirely. I think I'd make the React stuff not `PUT` back the value, which is fine. I can do this.

But for optimization, I don't think there'd really be one. Even if we decided to skip writing the value to the database, we still have to query the database to check if there is an override. So no matter what, not sure we could do any fast path to skipping and only reading value from disk. This is where the local caching for Options helps though. No matter what, we still have to query the db though to check if it's there. :(